### PR TITLE
Appearance (Shader, Texture), Physical, and Thermal Assets

### PR DESCRIPTION
**Pull request by eirannejad**
Implements:
- [Generic Appearance Asset](https://apidocs.co/apps/revit/2020/dd16eb59-16ec-f121-289b-a69d26d7c789.htm)
- [Unified Bitmap Texture Asset](https://apidocs.co/apps/revit/2020/de22f405-e0d8-e50a-096f-7e199c64fd00.htm)
- [Checker Texture Asset](https://apidocs.co/apps/revit/2020/ff4ecb34-6fef-402a-5cee-c7937974b8d2.htm)
- [Physical Asset](https://apidocs.co/apps/revit/2020/28d74a77-581d-d4c5-7970-55f298eb85f7.htm)
- [Thermal Asset](https://apidocs.co/apps/revit/2020/cad9dcdd-0c26-e960-9646-c23349dc6ab5.htm)

For Revit >=2019

### CODE REVIEW

**Review by kike-garbo — COMMENTED** [review]
Changed review comments.

File: src/RhinoInside.Revit.GH/Components/Element/Material/BaseAssetComponent.cs [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+        case DB.Visual.AssetPropertyTime timeProp:
+          return timeProp.Value;
+
+        case DB.Visual.AssetPropertyUInt64 uint64Prop:
+          return uint64Prop.Value;
+      }
+      return null;
+    }
+
+    public static List<DB.Visual.Asset> GetLibrayAssets(DB.Visual.AssetType assetType)
```
[the remark this answers is not in the document]
Comment (eirannejad) [reply src/RhinoInside.Revit.GH/Components/Element/Material/BaseAssetComponent.cs:384]: Yup. Wrapped the Revit API so we can potentially change how this works internally in the future. Currently only works on Appearance assets.

File: src/RhinoInside.Revit/External/DB/Extensions/Document.cs [hunk 2]
```diff
@@ -410,6 +410,19 @@ public static bool GetPurgableElementTypes(this Document doc, out ICollection<El
...
+      => AppearanceAssetElement.GetAppearanceAssetElementByName(doc, name);
+
+    public static PropertySetElement FindPropertySetElement(this Document doc, string name)
+    {
+      return new FilteredElementCollector(doc)
+                 .OfClass(typeof(PropertySetElement))
+                 .ToElements()
+                 .Cast<PropertySetElement>()
+                 .Where(x => x.GetParameterValue<string>(BuiltInParameter.PROPERTY_SET_NAME) == name)
+                 .FirstOrDefault();
```
[the remark this answers is not in the document]
Comment (eirannejad) [reply src/RhinoInside.Revit/External/DB/Extensions/Document.cs:425]: Uh nice. Didn't know `WhereParameterEqualsTo` is there. So easy to use

**Comment by eirannejad** [comment]
Don't revert the changes on https://github.com/mcneel/rhino.inside-revit/pull/326/commits/52506e05c0abc6b7b76e79c81592517b24f1966f 
- `Physical/Thermal Asset Behaviour` is specific to `Physical` and `Thermal` assets. It is the behaviour of the real-world physical material and not Revit's Material (Previous value of `Material Behaviour` is confusing and also is badly sorted in between the components)
- Renamed the other value pickers to `Physical Asset Type` and `Thermal Asset Type` for consistency.
- They all get sorted at the bottom of the components list in GH UI.

**Pull request merged**